components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java

```diff
@@ -127,6 +127,15 @@ public String buildIDToken(OAuthTokenReqMessageContext tokenReqMsgCtxt,
 
         AuthenticatedUser authorizedUser = tokenReqMsgCtxt.getAuthorizedUser();
         String subjectClaim = getSubjectClaim(tokenReqMsgCtxt, tokenRespDTO, clientId, spTenantDomain, authorizedUser);
+        // Get subject identifier according to the configured subject type.
+        String callbackUri;
+        try {
+            callbackUri = OIDCClaimUtil.getCallbackUrl(clientId);
+        } catch (InvalidOAuthClientException e) {
+            throw new IdentityOAuth2Exception("Error while getting the registered callback URI for client_id: " +
+                    clientId, e);
```

Comment:
Since the caught exception is a client exception, lets throw an IdentityOAuth2ClientException which seems to be the client type exception for IdentityOAuth2Exception